knowledge-base/adr/0028-public-api-extraction.md

```diff
@@ -52,21 +56,35 @@ Hence, we propose the following approach:
      This removes the internal API form the `d.ts` files.
   2. Maintain minimal named exports in the root `index.ts` pointing to the objects of the public API.
      This creates also minimal module exports in the JavaScript usecase.
+  3. We create a `internal` export as we did for the [v4 case](https://github.com/SAP/cloud-sdk-js/tree/v1.28.0/packages/core) which can be imported via `@sap-cloud-sdk/odata-v2/internal`.
+     This internal package contains all exports.
 
 This is a large manual effort initially and seems to be redundant because the minimal `index.ts` alone would already do the trick.
 However, the double maintenance makes better check rules possible.
 We plan to implement the following checks:
 
 - Have a check to avoid any `*` exports in the root `index.ts`.
-- Have a check to enforce a single `index.ts` on root level specifying all public exports.
+- Use an automatic tool like [barrelsby](https://github.com/bencoveney/barrelsby#readme) to create `index.ts` exporting everything to the `internal` folder.
 - Have a check to enforce exposed object from `d.ts` match the named exports of the root `index.ts`.
   - A missing value in the `index.ts` denotes: You have exported something in the code but not added it to the relevant barrel or missed the `@internal`.
   - A missing value in the `d.ts` denotes: You have violated the API contract and marked a previously exported object as `@internal`.
+- In the integration and E2E tests we enforce root level imports via eslint rule.
```

Comment:
Clarification: Should the vdm e2e tests also have only root level imports? As there are a couple of imports from `dist` in the template apps as well. If not, then we could use with the internal, e.g. `@sap-cloud-sdk/odata-v2/internal`

Reply:
I think the client side so the call to the backend-app should not have any internal imports. The backend side does not really matter.